micro-rdk/src/esp32/entry.rs

```diff
@@ -142,6 +156,136 @@ pub async fn serve_web_inner(
     srv.serve(robot).await;
 }
 
+#[cfg(feature = "provisioning")]
+async fn serve_provisioning_async<S>(
+    ip: Ipv4Addr,
+    exec: Esp32Executor,
+    info: ProvisioningInfo,
+    storage: S,
+    last_error: Option<String>,
+) -> Result<(AppClientConfig, Esp32TLSServerConfig), Box<dyn std::error::Error>>
+where
+    S: CredentialStorage + Clone + 'static,
+    S::Error: Debug,
+    GrpcError: From<S::Error>,
+{
+    use super::conn::mdns::Esp32Mdns;
+    use std::ffi::CString;
+    let _ = Timer::after(std::time::Duration::from_millis(150)).await;
+    let hostname = format!(
+        "provisioning-{}-{}",
+        info.get_model(),
+        info.get_manufacturer()
+    );
+    let mut mdns = Esp32Mdns::new(hostname)?;
+
+    let srv = ProvisioningServiceBuilder::new().with_provisioning_info(info);
+
+    let srv = if let Some(error) = last_error {
+        srv.with_last_error(error)
+    } else {
+        srv
+    };
+    let srv = srv.build(storage.clone());
+    let listen = TcpListener::bind(ip.to_string() + ":0")?;
+    let listen: Async<TcpListener> = listen.try_into()?;
+
+    let port = listen.get_ref().local_addr()?.port();
+
+    mdns.add_service(
+        "provisioning",
```

Comment:
If this string is meant to be the same between native and esp32, maybe it should be a constant somewhere. Ditto other uses of the `provisioning` string. Also, see my comment in the native version of this about the exact contentx of the string.